Components/Core/Timer.cpp

```diff
@@ -20,15 +25,55 @@ Timer::Timer()
 	// We make a timer named "Timer" with a callback function that does nothing, Autoreload false, and the default period of 1s.
 	// The timer ID is specified as (void *)this to provide a unique ID for each timer object - however this is not necessary for polling timers.
 	// The timer is created in the dormant state.
-	rtTimerHandle = xTimerCreate("Timer", DEFAULT_TIMER_PERIOD, pdFALSE, (void *)this, empty_callback);
+	assert (rtTimerHandle = xTimerCreate("Timer", DEFAULT_TIMER_PERIOD, pdFALSE, (void *)this, empty_callback));
+	timerState = UNINITIALIZED;
 }
 
 /**
  * @brief Changes this timer object's RTOS timer period, returns true on success, returns false on failure (timer command queue full)
  */
 bool Timer::ChangePeriod(const uint32_t period)
```

Comment:
Please make two different functions, this one should be something like **ChangePeriodAndStart(const uint32_t period_ms)**

And the other shouldn't start the timer (or immedietly stop it) just called ChangePeriod

Reply:
What should the ChangePeriod function contain? Invoking xTimerChangePeriod automatically starts the timer so from what I understand we should convert ms to ticks in the change period function.

Reply:
Whatever is in ChangePeriodAndStart, just stop it immediately after (and set the internal state to PAUSED instead). You can actually just call the StopTimer() function from within this function.